samples/web/content/apprtc/apprtc.py

```diff
@@ -495,12 +515,109 @@ def write_response(self, result, params, messages):
+      self.report_error(constants.RESPONSE_INVALID_CALLER)
+      return
+    caller = callerRecords[0]
+    # Look up callee by id.
+    calleeRecords = gcmrecord.GCMRecord.get_by_user_id(calleeId, True)
+    if len(calleeRecords) < 1:
+      self.report_error(constants.RESPONSE_INVALID_CALLEE)
+      return
+
+    if does_room_exist(self.request.host_url, room_id):
+      self.report_error(constants.RESPONSE_INVALID_ROOM)
+      return
+    result = add_client_to_room(
+        self.request.host_url, room_id, caller.user_id, False, calleeId)
+    if result['error'] is not None:
+      logging.info('Error adding client to room: ' + result['error'] + \
+          ', room_state=' + result['room_state'])
+      self.write_response(result['error'], {}, [])
+      return
+    
+    # TODO (chuckhays): Initiate message to verified gcm ids to ring.
+    
+    self.write_room_parameters(
+        room_id, caller.user_id, result['messages'], result['is_initiator'])
+      
+    logging.info('User ' + caller.user_id + ' initiated call to ' + calleeId + \
+        ' from gcmId ' + callerId)
+    logging.info('User ' + caller.user_id + ' joined room ' + room_id)
+    logging.info('Room ' + room_id + ' has state ' + result['room_state'])
+
+  def handle_accept(self, msg, room_id):
+    """Handles a callee accepting a call request."""
+    if not util.has_msg_field(msg, PARAM_CALLEE_GCM_ID, basestring):
+      self.report_error(constants.RESPONSE_INVALID_ARGUMENT)
+      return
+
+    calleeId = msg[PARAM_CALLEE_GCM_ID]
+    # Look up and validate callee by gcm id.
+    calleeRecords = gcmrecord.GCMRecord.get_by_gcm_id(calleeId, True)
+    if len(calleeRecords) < 1:
+      self.report_error(constants.RESPONSE_INVALID_CALLEE)
+      return
+
+    callee = calleeRecords[0]
+    
+    if not does_room_exist(self.request.host_url, room_id):
+      self.report_error(constants.RESPONSE_INVALID_ROOM)
+      return
+
+    result = add_client_to_room(
+        self.request.host_url, room_id, callee.user_id, False)
+    if result['error'] is not None:
+      logging.info('Error adding client to room: ' + result['error'] + \
+          ', room_state=' + result['room_state'])
+      self.write_response(result['error'], {}, [])
+      return
+    
+    # TODO (chuckhays): Send message to ringing clients to stop ringing.
+    
+    self.write_room_parameters(
+        room_id, callee.user_id, result['messages'], result['is_initiator'])
+      
+    logging.info('User ' + callee.user_id + ' accepted call ' + \
+        'from gcmId ' + calleeId)
+    logging.info('User ' + callee.user_id + ' joined room ' + room_id)
+    logging.info('Room ' + room_id + ' has state ' + result['room_state'])
 
   def post(self, room_id):
+    # Check request body to determine what action to take.
+    msg = util.get_message_from_json(self.request.body)
+    if util.has_msg_field(msg, PARAM_ACTION, basestring):
+      action = msg[PARAM_ACTION]
+      if action == 'call':
```

Comment:
@jiayliu did you have something in mind for how to differentiate these calls? I just went with what was simple at the time (including a action: value) because it wasn't clear to me if we wanted to be specific or just try to infer from the values passed.